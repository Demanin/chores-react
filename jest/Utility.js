import crypto from 'crypto';

const getRandomString = (length = 20) => {
  return crypto.randomBytes(length).toString('hex');
};

const getRandomNumber = (min = 0, max = 9999999) => {
  min = Math.ceil(min);
  max = Math.floor(max);

  return Math.floor(Math.random() * (max - min)) + min;
};

const getRandomBool = () => {
  return !!getRandomNumber(0, 1);
};

export default {
  getRandomString,
  getRandomNumber,
  getRandomBool,
};
