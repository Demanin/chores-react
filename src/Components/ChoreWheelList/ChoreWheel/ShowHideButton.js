import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import PropTypes from 'prop-types';

class ShowHideButton extends Component
{
  render() {
    if (this.props.choreWheel.isVisible) {
      return (
        <Button
          variant="link"
          onClick={() => this.props.onHideClick(this.props.choreWheel.id)}
        >
          <FontAwesomeIcon icon="eye" />
        </Button>
      );
    }

    return (
      <Button
        variant="link"
        onClick={() => this.props.onShowClick(this.props.choreWheel.id)}
      >
        <FontAwesomeIcon icon="eye-slash" />
      </Button>
    );
  }
}

ShowHideButton.propTypes = {
  choreWheel: PropTypes.shape({
    isVisible: PropTypes.bool,
    id: PropTypes.number,
  }),
  onHideClick: PropTypes.func,
  onShowClick: PropTypes.func,
};

export default ShowHideButton;
