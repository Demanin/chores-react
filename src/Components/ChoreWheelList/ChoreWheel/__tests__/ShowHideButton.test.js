import React from 'react';
import ShowHideButton from 'src/Components/ChoreWheelList/ChoreWheel/ShowHideButton';
import Utility from 'jest/Utility';
import { shallow } from 'enzyme';

describe('ShowHideButton', () => {
  let onShowClick;
  let onHideClick;

  beforeEach(() => {
    onShowClick = jest.fn();
    onHideClick = jest.fn();
  });

  test('hidden snapshot', () => {
    const fixture = shallow(<ShowHideButton
      choreWheel={{ isVisible: false }}
      {...{ onShowClick, onHideClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('visible snapshot', () => {
    const fixture = shallow(<ShowHideButton
      choreWheel={{ isVisible: true }}
      {...{ onShowClick, onHideClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('show click called', () => {
    const id = Utility.getRandomNumber();

    const fixture = shallow(<ShowHideButton
      choreWheel={{ isVisible: false, id}}
      {...{ onShowClick, onHideClick }}
    />);

    fixture.simulate('click');

    expect(onShowClick).toHaveBeenCalledWith(id);
    expect(onHideClick).not.toHaveBeenCalled();
  });

  test('hide click called', () => {
    const id = Utility.getRandomNumber();

    const fixture = shallow(<ShowHideButton
      choreWheel={{ isVisible: true, id}}
      {...{ onShowClick, onHideClick }}
    />);

    fixture.simulate('click');

    expect(onHideClick).toHaveBeenCalledWith(id);
    expect(onShowClick).not.toHaveBeenCalled();
  });
});
