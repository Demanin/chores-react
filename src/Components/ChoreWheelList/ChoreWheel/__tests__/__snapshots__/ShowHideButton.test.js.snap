// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowHideButton hidden snapshot 1`] = `ShallowWrapper {}`;

exports[`ShowHideButton visible snapshot 1`] = `ShallowWrapper {}`;
